### ISSUE TITLE
Info: Signal-to-influxdb - successful setup with influxdb v2 and Raspian Bullseye 64-bit

### ISSUE THREAD

**Issue by pbfulmar** [tkurki/signalk-to-influxdb#61]
Dear Teppo,
just wanted to let you know, that I have my whole SignalK setup now running on Raspi 64-Bit and influxdb v2.
With influxdb v1 the CPU usage recently was ramping up to more than 90%, now the system is only using 16% and that is absolutely stable since 6 days.
I‘m using the v1 API of influxdb v2, no modification to the SignalK influxdb plugin.

Let me know if you like to see the details of the influxdb setup.

Cheers
Peter
En route, SY Joy

**Comment by miggat** [comment]
I know it's been some time, but i'd be certainly interested in knowing how you acheived it. I cannot find a way to connect the plugin to influx v2.

**Comment by leeprevost** [comment]
@pbfulmar would also like to know how you did it.  I'd be willing to help get this upgraded to v2 as there are several reasons I want to go there and you just gave me another one (stable/cpu usage).  My reason is that I want to do some local processing of datapoints as they come in and I can't do math across measurements without flux an v2 as I understand it.   I've upgraded my Rasipi to bullseye and signalk seems to b performing fine.  But, I have found the other influx add on to be insufficient as it really doesn't seem to support a localhost db (ie. informdb oss ver 2+).   

I'm a python guy and no expert on Node but it looks like:

1. This plug-in uses [node-influx](https://node-influx.github.io/) which support ver 1.x
2. The newer javascript db driver supports 1.8+ an all 2+ instances.   That is [influxdb-client-js](https://influxdata.github.io/influxdb-client-js/influxdb-client.influxdb.html).   It looks like very liittle changes are needed on the connection instance.  But some changes needed on the methods.

Old client method -- influx.writePoints, influx.query
New client method - influx.getWriteApi, influx.getQueryApi

Also, not sure whether tokens and organizations are needed for local/OSS mode.  Most of the refactor seems to be in the index.js file in /src where the client is configured.   

If you would be willing to share, I could fork this and then post a pull-request for the author to update?

Cheers!

Lee

**Comment by amirlanesman** [comment]
@pbfulmar Hey please share how you got it working with Influxdb2. I am having difficulties getting it to work.

**Comment by amirlanesman** [comment]
Just for anyone else going through here:

Found this article which I guess is what @pbfulmar did. In any case it worked for me, followed the article exactly (changed names and values obviously) and it worked great and didn't require any changes to the plugin itself.

https://www.techetio.com/2021/11/29/influxdb-v2-using-the-v1-api-for-v1-dependent-applications/

TL;DR:

execute the following on the influxdb machine
``
token=RCU5A_a1W2qrMlMO1...
org=Organization
bucket_id=a0a0ac9ddd0e716e
influx user create --name signalk --password mypassword --org $org --token $token
influx v1 dbrp create --db boatdata --bucket-id $bucket_id --rp Forever --org $org --token $token
influx v1 auth create --username signalk --read-bucket $bucket_id --write-bucket $bucket_id --password mypassword --org $org --token $token
```

**Comment by leeprevost** [comment]
@amirlanesman - that was a big help.  Worked like a champ!   Journal dump:

```
Jul 28 18:57:49 raspberrypi systemd[1]: signalk.service: Succeeded.
Jul 28 18:57:49 raspberrypi systemd[1]: signalk.service: Consumed 7.440s CPU time.
Jul 28 18:57:49 raspberrypi systemd[1]: signalk.service: Scheduled restart job, restart counter is at 2.
Jul 28 18:57:49 raspberrypi systemd[1]: Stopped "Starting signalk server".
Jul 28 18:57:49 raspberrypi systemd[1]: signalk.service: Consumed 7.440s CPU time.
Jul 28 18:57:49 raspberrypi systemd[1]: Started "Starting signalk server".
Jul 28 18:57:52 raspberrypi signalk-server[2641]: *** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
Jul 28 18:57:52 raspberrypi signalk-server[2641]: *** WARNING *** Please fix your application to use the native API of Avahi!
Jul 28 18:57:52 raspberrypi signalk-server[2641]: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
Jul 28 18:57:52 raspberrypi signalk-server[2641]: *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
Jul 28 18:57:52 raspberrypi signalk-server[2641]: *** WARNING *** Please fix your application to use the native API of Avahi!
Jul 28 18:57:52 raspberrypi signalk-server[2641]: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
Jul 28 18:57:52 raspberrypi signalk-server[2641]: signalk-server running at 0.0.0.0:[object Object]
Jul 28 18:57:54 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:54 +0000] "GET /admin/ HTTP/1.1" 200 562 "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:54 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:54 +0000] "GET /admin/main.js HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:54 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:54 +0000] "GET /@signalk/vesselpositions/remoteEntry.js HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:54 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:54 +0000] "GET /admin/436.js HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:54 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:54 +0000] "GET /admin/784.js HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:54 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:54 +0000] "GET /admin/270.js HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:54 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:54 +0000] "GET /admin/316.js HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /skServer/plugins HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /skServer/loginStatus HTTP/1.1" 200 201 "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /skServer/webapps HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /skServer/addons HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /signalk HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /admin/fonts/Simple-Line-Icons.3826fa1cb2348dd93948a50cbd2b8fb6.woff2 HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /skServer/security/access/requests HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /admin/fonts/signal-k-logo-image-text.d132ec2c4f0269c103c868aaa6868f01.svg HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /admin/img/favicon.ico HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:55 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:55 +0000] "GET /admin/fonts/fa-solid-900.1551f4f60c37af51121f106501f69b80.woff2 HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 18:57:56 raspberrypi signalk-server[2641]: ::ffff:192.168.1.26 - - [28/Jul/2022:22:57:56 +0000] "GET /skServer/appstore/available HTTP/1.1" 304 - "http://raspberrypi.lan/admin/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
Jul 28 19:00:02 raspberrypi signalk-server[2641]: TcpProvider:connect ETIMEDOUT 192.168.4.1:1457
**Jul 28 19:01:10 raspberrypi systemd[1]: Stopping "Starting signalk server"...**
Jul 28 19:01:10 raspberrypi systemd[1]: signalk.service: Succeeded.
Jul 28 19:01:10 raspberrypi systemd[1]: Stopped "Starting signalk server".
Jul 28 19:01:10 raspberrypi systemd[1]: signalk.service: Consumed 5.311s CPU time.
**Jul 28 19:01:11 raspberrypi systemd[1]: /etc/systemd/system/signalk.service:7: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removing the setting altogether.**
**Jul 28 19:01:11 raspberrypi systemd[1]: /etc/systemd/system/signalk.service:8: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removing the setting altogether.**
Jul 28 19:01:11 raspberrypi systemd[1]: /etc/systemd/system/signalk.service:7: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removing the setting altogether.
Jul 28 19:01:11 raspberrypi systemd[1]: /etc/systemd/system/signalk.service:8: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removing the setting altogether.
Jul 28 19:01:11 raspberrypi systemd[1]: signalk.socket: Succeeded.
Jul 28 19:01:11 raspberrypi systemd[1]: Closed signalk.socket.
Jul 28 19:01:14 raspberrypi systemd[1]: Listening on signalk.socket.
Jul 28 19:03:04 raspberrypi systemd[1]: Started "Starting signalk server".
Jul 28 19:03:09 raspberrypi signalk-server[1025]: *** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
Jul 28 19:03:09 raspberrypi signalk-server[1025]: *** WARNING *** Please fix your application to use the native API of Avahi!
Jul 28 19:03:09 raspberrypi signalk-server[1025]: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
Jul 28 19:03:09 raspberrypi signalk-server[1025]: *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
Jul 28 19:03:09 raspberrypi signalk-server[1025]: *** WARNING *** Please fix your application to use the native API of Avahi!
Jul 28 19:03:09 raspberrypi signalk-server[1025]: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
Jul 28 19:03:10 raspberrypi signalk-server[1025]: signalk-server running at 0.0.0.0:[object Object]
```
Maybe I've inadvertenly added a scraper?  Also a problem with derived data on one of my engines?

The only thing that may look abnormal but would need someone more knowledgeable than me to verify:
1) A problem with the unit file "syslog is obsolete"

Also, later, in journalctl, I see a few db errors but I think it may have been when I shutdown or very specific to the derived data plug-in.  Will test these later:

```
Jul 28 19:35:38 raspberrypi signalk-server[747]: signalk-to-influxdb:Error: write EPIPE
Jul 28 19:35:38 raspberrypi signalk-server[747]: Error: write EPIPE
Jul 28 19:35:38 raspberrypi signalk-server[747]:     at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:94:16)
Jul 28 19:35:44 raspberrypi signalk-server[747]: signalk-to-influxdb:Error: No host available
Jul 28 19:35:44 raspberrypi signalk-server[747]: Error: No host available
Jul 28 19:35:44 raspberrypi signalk-server[747]:     at Pool.stream (/home/pi/.signalk/node_modules/influx/lib/src/pool.js:235:29)
Jul 28 19:35:44 raspberrypi signalk-server[747]:     at /home/pi/.signalk/node_modules/influx/lib/src/pool.js:166:18
Jul 28 19:35:44 raspberrypi signalk-server[747]:     at new Promise (<anonymous>)
Jul 28 19:35:44 raspberrypi signalk-server[747]:     at Pool.discard (/home/pi/.signalk/node_modules/influx/lib/src/pool.js:165:16)
Jul 28 19:35:44 raspberrypi signalk-server[747]:     at InfluxDB.writePoints (/home/pi/.signalk/node_modules/influx/lib/src/index.js:854:27)
Jul 28 19:35:44 raspberrypi signalk-server[747]:     at /home/pi/.signalk/node_modules/signalk-to-influxdb/built/index.js:339:43
Jul 28 19:35:44 raspberrypi signalk-server[747]:     at runMicrotasks (<anonymous>)
Jul 28 19:35:44 raspberrypi signalk-server[747]:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

On the influx journal, I see a few problems.  Maybe a 

```
Jul 28 20:18:59 raspberrypi influxd-systemd-start.sh[644]: ts=2022-07-29T00:18:59.691579Z lvl=error msg="Unable to gather" log_id=0bym5oo0000 service=scraper scraper-name="new target" error="Get \"http://raspberrypi.lan:8086/metrics\": dial tcp: lookup raspberrypi.lan on 192.168.4.1:53: read udp 192.168.4.3:55544->192.168.4.1:53: read: connection refused"
Jul 28 20:19:09 raspberrypi influxd-systemd-start.sh[644]: ts=2022-07-29T00:19:09.713829Z lvl=error msg="Unable to gather" log_id=0bym5oo0000 service=scraper scraper-name="new target" error="Get \"http://raspberrypi.lan:8086/metrics\": dial tcp: lookup raspberrypi.lan on 192.168.4.1:53: read udp 192.168.4.3:43634->192.168.4.1:53: read: connection refused"
Jul 28 20:19:12 raspberrypi signalk-server[747]: signalk-to-influxdb:Error: A 400 Bad Request error occurred: {"code":"invalid","message":"unable to parse 'propulsion.starboard.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053950159000000': invalid boolean\nunable to parse 'propulsion.port.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053950494000000': invalid boolean\nunable to parse 'propulsion.starboard.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053950495000000': invalid boolean\nunable to parse 'propulsion.port.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053950791000000': invalid boolean\nunable to parse 'propulsion.starboard.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053950792000000': invalid boolean\nunable to parse 'propulsion.port.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053951046000000': invalid boolean\nunable to parse 'propulsion.starboard.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053951047000000': invalid boolean\nunable to parse 'propulsion.port.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053951304000000': invalid boolean\nunable to parse 'propulsion.starboard.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053951304000000': invalid boolean\nunable to parse 'propulsion.port.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053951521000000': invalid boolean\nunable to parse 'propulsion.starboard.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053951521000000': invalid boolean\nunable to parse 'propulsion.port.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053951742000000': invalid boolean\nunable to parse 'propulsion.starboard.fuel.economy,context=vessels.urn:mrn:imo:mmsi:338374822,source=derived-data value=Infinity 1659053951742000000': invalid boolean"}
Jul 28 20:19:12 raspberrypi signalk-server[747]:     at IncomingMessage.<anonymous> (/home/pi/.signalk/node_modules/influx/lib/src/pool.js:50:38)
```

Finally, I do hope the author rings in here. I think a better solution may be a slight refactor that supports the newer v2 client.  I'm willing to try to fork it and put in a pull request but would like a little guidance on what parts of code i need to focus on. I assume its here: [/src/skToInflux.js](https://github.com/tkurki/signalk-to-influxdb/blob/main/src/skToInflux.js#L146)

**Comment by leeprevost** [comment]
One other slight tweak.  The startup of services and sequencing of signalk, influx, and grafana seem to be important.  I get db errors if signalk starts up before the influxdb.   To resolve this, I added a unit definition in: /etc/systemd/system/signalk.service as follows:

```
[Unit]
Description="Starting signalk server"
After=multi-user.target
```
this makes signalk wait until after the others are started and the errors go away.

**Comment by pbfulmar** [comment]
Hi guys,
sorry for replying so late. At times, as a full time cruiser you're quite occupied ;-)

I just took a quick glimpse at the above mentioned article. Seems thats mostly what I did.
All the required information can also be found in the online documentation of influx.
Here is a summary of my steps for your reference. I hope I did not introduce some errors when writing this summary, be cautious.

- initial setup, you can run this only once:
    influx setup \
  	   --org example-org \
           --bucket example-bucket \
           --username <your-user> \
           --password <your-password> \
           --force
  Basically, I was just using the useraccount later, but not the example-org and example-bucket.
  You can clean this up with the influx command, later.
  
- create org and bucket:
    influx org create --name <your-org>
    influx bucket create -o <your-org> --name <your-bucket>  

- find out your bucket ids:
    influx bucket list -o <your-org>

- create access for v1 api, using the bucket ids from the previous command:
    influx v1 auth create -o <your-org> \
           --username <your-user> \
	   --write-bucket <your-bucket-id> --read-bucket <your-bucket-id>
  You have to supply the password of <your-user>

- create retention policy (using the default retention policy):
    influx v1 dbrp create \
           -o <your-org> --bucket-id <your-bucket-id> \
	   --db <your-db> --rp <your-rp>  --default

- check result:
  influx v1 dbrp list -o <your-org>

To configure the signalk-to-influxdb plugin of signalk, you need to supply
the <your-user>, <your-password> and <your-db> from the above commands. 

Cheers
Peter
En route, SY Joy

**Comment by pbfulmar** [comment]
Hi guys,
sorry for replying so late. At times, as a full time cruiser you're quite occupied ;-)

I just took a quick glimpse at the above mentioned article. Seems thats mostly what I did.
All the required information can also be found in the online documentation of influx.
Here is a summary of my steps for your reference. I hope I did not introduce some errors when writing this summary, be cautious.

- initial setup, you can run this only once:
`influx setup --org example-org --bucket example-bucket --username "your-user" --password "your-password" --force`

  Basically, I was just using the useraccount later, but not the example-org and example-bucket.
  You can clean this up with the influx command, later.
 
- create org and bucket:
    `influx org create --name "your-org"`
    `influx bucket create -o "your-org" --name "your-bucket"`  

- find out your bucket ids:
    `influx bucket list -o "your-org"`

- create access for v1 api, using the bucket ids from the previous command:
    `influx v1 auth create -o "your-org" 
           --username "your-user" 
	   --write-bucket "your-bucket-id" --read-bucket "your-bucket-id"`
  You have to supply the password of "your-user"

- create retention policy (using the default retention policy):
    'influx v1 dbrp create 
           -o "your-org" --bucket-id "your-bucket-id"
	   --db "your-db" --rp "your-rp" --default`

- check result:
  `influx v1 dbrp list -o "your-org"`

To configure the signalk-to-influxdb plugin of signalk, you need to supply
the "your-user", "your-password" and "your-db" from the above commands. 

Cheers
Peter
En route, SY Joy

**Comment by pbfulmar** [comment]
Hi guys,
sorry for replying so late. At times, as a full time cruiser you're quite occupied ;-)

I just took a quick glimpse at the above mentioned article. Seems thats mostly what I did.
All the required information can also be found in the online documentation of influx.
Here is a summary of my steps for your reference. I hope I did not introduce some errors when writing this summary, be cautious.

- initial setup, you can run this only once:
`influx setup --org example-org --bucket example-bucket --username "your-user" --password "your-password" --force`

  Basically, I was just using the useraccount later, but not the example-org and example-bucket.
  You can clean this up at any time with the influx command.
 
- create org and bucket:
    `influx org create --name "your-org"`
    `influx bucket create -o "your-org" --name "your-bucket"`  

- find out your bucket id:
    `influx bucket list -o "your-org"`

- create access for v1 api, using the bucket id from the previous command:
    `influx v1 auth create -o "your-org" 
           --username "your-user" 
	   --write-bucket "your-bucket-id" --read-bucket "your-bucket-id"`
  You have to supply the password of "your-user".

- create retention policy (using the default retention policy):
    `influx v1 dbrp create 
           -o "your-org" --bucket-id "your-bucket-id"
	   --db "your-db" --rp "your-rp" --default`

- check result:
  `influx v1 dbrp list -o "your-org"`

To configure the signalk-to-influxdb plugin of signalk, you need to supply
the "your-user", "your-password" and "your-db" from the above commands. 

Cheers
Peter
En route, SY Joy

**Comment by leeprevost** [comment]
Much thanks Peter.  Yes, this was similar to the approach @amirlanesman found.   

I appreciate you pulling into a port and solving a mystery for us.   :)

Your full time cruiser status reminds me of Travis McGee.   Here in the US, author John McDonald created the fictitious character "Travis McGee" who traveled abord the "Busted Flush" throughout Florida.  He was a retired police detective who would solve a mystery in every port while "taking his retirement in installments."  

[Legend: Travis McGee and the Busted Flush](https://www.visitflorida.com/travel-ideas/articles/boating-travis-mcgee-and-the-busted-flush-legend/)

**Comment by tkurki** [comment]
@leeprevost we could change the server setup script to include an option to specify/pick the services that SK depends on. So an option in the setup script that could add explicit dependency to `influxdb.service` in the unit file. This is not really influxdb specific.

**Comment by tkurki** [comment]
I created a separate issue for real InfluxDb 2 support #63.
